src/lib/ts/tsconfig.ts

```diff
@@ -44,22 +44,39 @@ export const initializeTsConfig = (defaultTsConfig: TsConfig, entryPoint: NgEntr
   const basePath = path.dirname(entryPoint.entryFilePath);
 
   // Resolve defaults from DI token and create a deep copy of the defaults
-  const tsConfig = JSON.parse(JSON.stringify(defaultTsConfig));
+  let tsConfig: TsConfig = JSON.parse(JSON.stringify(defaultTsConfig));
 
-  tsConfig.rootNames = [entryPoint.entryFilePath];
-  tsConfig.options.flatModuleId = entryPoint.moduleId;
-  tsConfig.options.flatModuleOutFile = `${entryPoint.flatModuleFile}.js`;
-  tsConfig.options.basePath = basePath;
-  tsConfig.options.baseUrl = basePath;
-  tsConfig.options.rootDir = basePath;
-  tsConfig.options.outDir = '';
-  // setting this as basedir will rewire triple-slash references
-  tsConfig.options.declarationDir = basePath;
+  // minimal compilerOptions needed in order to avoid errors, with their associated default values
+  // some are not overrided in order to keep the default associated TS errors if the user choose to set incorrect values
+  const requiredOptions: Partial<ng.CompilerOptions> = {
```

Comment:
In order to do this, I used https://github.com/dherges/ng-packagr/blob/master/src/lib/ts/conf/tsconfig.ngc.json to set https://github.com/angular/devkit/pull/750/files#diff-fce66255aa6a3ef162bcacc08960d5fcR1 and then removed on parameter after the other to see if an error occured.

I you think more centralisation is needed, I may "import" https://github.com/dherges/ng-packagr/blob/master/src/lib/ts/conf/tsconfig.ngc.json or, if their is already a centralisation point, please tell me.